### PR TITLE
feat: handle subtitles for video streaming

### PR DESCRIPTION
**Pull request by VLEFF**
#460 

Hello there,
Here my little contribution to this project =)

Frontend part is on his own PR : https://github.com/filebrowser/frontend/pull/18

I hope my code fulfill your requirements, if not, I'll change what's needed =) 

To test it : Just have .srt or .vtt file in the same folder of your video, if your browser supports html5 track tag, it will be automatically added

Html5 standard and only recognized subtitle format is WebVTT (.vtt). 
Since .vtt and .srt are nearly constructed in the same manner, and that .srt is way more used than .vtt, I have handled a simple srt to vtt conversion (replace coma in times with dot and add WEBVTT header to the top of the file)

### CODE REVIEW

File: http/subtitle.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	str := string(b) // convert content to a 'string'
+	ext := filepath.Ext(filename)
+	if(ext == ".srt") {
+        	re := regexp.MustCompile("([0-9]{2}:[0-9]{2}:[0-9]{2}),([0-9]{3})")
+        	str = "WEBVTT\n\n" + re.ReplaceAllString(str, "$1.$2")
+	}
+	return str, err
+} 
+
+func ReadDir(dirname string) ([]os.FileInfo, error) {
```
Comment (VLEFF) [inline http/subtitle.go]: Maybe this function is something you already have in some util lib?

Comment (hacdias) [reply http/subtitle.go]: yeah, maybe, but I can't see where. Let's keep it!

**Review by hacdias — COMMENTED** [review]
On overall, it seems very good! I don't know why CirleCI didn't run on this PR, but if it did, it would have fail: indentation! Could you run `gofmt -w *.go`?

**Comment by VLEFF** [comment]
gofmt done, didn't know this command... (actually I didn't know Go at all yesterday, but it's a secret =p)

**Comment by hacdias** [comment]
Thanks! Will release a new version soon then 😄

**Comment by kaueholt** [comment]
This feature doesn't work anymore? I tried in Mozila, Chrome and Opera, and neither of them displayed subtitle options or the subtitles. Am I missing something? Thanks!

**Comment by XDo0** [comment]
> This feature doesn't work anymore? I tried in Mozila, Chrome and Opera, and neither of them displayed subtitle options or the subtitles. Am I missing something? Thanks!

I have the same problem. Have you solved it? I tried in Edge for win11.

**Comment by kaueholt** [comment]
Hi. I have not.

**Pull request merged**